### PR TITLE
Add "Karma Inflation Adjusted" sorting option to All Posts tab

### PR DESCRIPTION
**Pull request by Will-Howard**

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 1]
```diff
@@ -262,6 +268,32 @@ const getFrontpageFilter = (filterSettings: FilterSettings): {filter: any, softF
...
+              $arrayElemAt: [
+                karmaInflationSeries.values,
+                {
+                  $max: [
+                    timeSeriesIndexExpr("$postedAt", karmaInflationSeries.start, karmaInflationSeries.interval),
+                    0 // fall back to first value if out of range
+                  ]
+                }]
+            },
+            karmaInflationSeries.values[-1] // fall back to final value if out of range
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/views.ts:289]: This just always returns undefined. In javascript `arr[-1]` does not return the last element of the array.

File: packages/lesswrong/components/users/UsersProfile.tsx [hunk 2]
```diff
@@ -230,14 +231,6 @@ const styles = (theme: ThemeType): JssStyles => ({
   },
 })
 
-const sortings: Partial<Record<string,string>> = {
-  magic: "Magic (New & Upvoted)",
-  recentComments: "Recent Comments",
-  new: "New",
-  old: "Old",
-  top: "Top"
-}
```
Comment (Will-Howard) [inline packages/lesswrong/components/users/UsersProfile.tsx:239]: this was a duplicate

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 3]
```diff
@@ -262,6 +268,32 @@ const getFrontpageFilter = (filterSettings: FilterSettings): {filter: any, softF
...
+            {
+              $arrayElemAt: [
+                karmaInflationSeries.values,
+                {
+                  $max: [
+                    timeSeriesIndexExpr("$postedAt", karmaInflationSeries.start, karmaInflationSeries.interval),
+                    0 // fall back to first value if out of range
+                  ]
+                }]
+            },
```
Comment (Will-Howard) [inline packages/lesswrong/lib/collections/posts/views.ts:288]: this is effectively using `baseScore / meanKarmaForMonth` as the field to sort by

File: packages/lesswrong/components/posts/PostsListSortDropdown.tsx [hunk 4]
```diff
@@ -41,25 +40,20 @@ const PostsListSortDropdown = ({classes, value}:{
...
+      {TAG_POSTS_SORT_ORDER_OPTIONS[value].label} <ArrowDropDownIcon className={classes.icon}/>
     </span>
     <Menu
       open={Boolean(anchorEl)}
       anchorEl={anchorEl}
       onClose={()=>setAnchorEl(null)}
     >
-        <QueryLink query={{sortedBy: undefined}} merge className={classes.menuItem}>
-          <MenuItem value={"relevance"} onClick={()=>setAnchorEl(null)}>
-            {TAG_POSTS_SORT_ORDER_OPTIONS["relevance"]}
```
Comment (Will-Howard) [inline packages/lesswrong/components/posts/PostsListSortDropdown.tsx:53]: this was unnecessary as all the SORT_ORDER_OPTIONS were already in TAG_POSTS_SORT_ORDER_OPTIONS

File: packages/lesswrong/lib/collections/posts/views.ts (hunk 1 above)
Comment (jpaddison3) [reply packages/lesswrong/lib/collections/posts/views.ts]: 🐍

**Pull request merged**